Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -1273,12 +1280,24 @@ void AppendPipeline::handleProtectedBufferProbeInformation(GstPadProbeInfo* info
 {
     ASSERT(GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_BUFFER);
 
+    bool wasProcessingProtectionEvents = m_isProcessingProtectionEvents;
     m_isProcessingProtectionEvents = false;
 
+    unsigned listSize = gst_value_list_get_size(&m_cachedProtectionEvents);
+    if (!listSize)
+        return;
+
+    if (wasProcessingProtectionEvents) {
+        GST_TRACE("sending %u protection events to the main thread", listSize);
+        GstStructure* structure = gst_structure_new_empty("demuxer-done-sending-protection-events");
+        gst_structure_set_value(structure, "stream-encryption-events", &m_cachedProtectionEvents);
+        GstMessage* message = gst_message_new_application(GST_OBJECT(m_demux.get()), structure);
+        gst_bus_post(m_bus.get(), message);
```

Comment:
Can we not do a runloop distpatch with demuxerIsDoneSendingProtectionEvents instead of going through the bus like this?

Reply:
Also, I'm sure there's still some protection event handling in the decryptors, can we delete that code now and handle them in just in the append pipeline?

Reply:
About the RunLoop, I prefer to keep it like this until @ntrrgc finishes her ATQ because all these things need to be abortable.

About removing the event handling in the decryptors, no we can't. Actually the session handling is there and what makes regular non-MSE playback work.